[[ Bug 22877 ]] Ensure mouse focus restored after relayer

montegoulding opened the pull request:
This patch ensures if the source object of the relayer command has the mouse
focus it is restored correctly afterwards.

runrevmark commented on engine/src/exec-interface2.cpp:
```diff
@@ -3494,7 +3494,7 @@ void MCInterfaceDoRelayer(MCExecContext& ctxt, int p_relation, MCObjectPtr p_sou
 		// Make sure we remove focus from the control.
```
I suggest augmenting this comment:
```
// Make sure we remove focus from the control.
// Note: Unlike KFOCUSED, the MFOCUSED flag is only set on the focused
// control and its (control) ancestors *not* the card.
```

montegoulding committed:
[[ Bug 22877 ]] Ensure mouse focus restored after relayer

This patch ensures if the source object of the relayer command has the mouse
focus it is restored correctly afterwards.

Previously the `CS_MFOCUSED` state of the card was being checked, however, this
is only ever set if `CS_MENU_ATTACHED`. This patch checks the `CS_MFOCUSED`
state of the source object instead.

runrevmark commented:
@livecode-vulcan review ok 7a8fd99

livecode-vulcan commented:
💙 review by @runrevmark ok 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

livecode-vulcan commented:
😞 test failure 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/859)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/268)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/268)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/268)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/268)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2385)
* try-community-universal-ios-iphoneos13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/37)
* try-community-universal-ios-iphonesimulator13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/36)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2414)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2401)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2400)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2435)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2375)

runrevmark commented:
@livecode-vulcan retry 7a8fd99

livecode-vulcan commented:
😞 test failure 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/860)
* try-community-armv7-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/269)
* try-community-arm64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/269)
* try-community-x86-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/269)
* try-community-x86_64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/269)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2386)
* try-community-universal-ios-iphoneos13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/38)
* try-community-universal-ios-iphonesimulator13.5: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/37)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2415)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2402)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2401)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2436)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2376)

livecodepanos commented:
@livecode-vulcan retry 7a8fd99

livecode-vulcan commented:
😞 test failure 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/861)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/270)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/270)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/270)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/270)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2387)
* try-community-universal-ios-iphoneos13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/39)
* try-community-universal-ios-iphonesimulator13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/38)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2416)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2403)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2402)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2437)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2377)

livecodepanos commented:
@livecode-vulcan retry 7a8fd99

livecode-vulcan commented:
😞 test failure 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/862)
* try-community-armv7-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/271)
* try-community-arm64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/271)
* try-community-x86-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/271)
* try-community-x86_64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/271)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2388)
* try-community-universal-ios-iphoneos13.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/40)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/39)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2417)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2404)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2403)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2438)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2378)

livecodepanos commented:
This PR keeps failing now that I re-activated build-mac-4, which runs Catalina. I'll disable build-mac-4 so as this PR will be checked by build-mac-3, that runs Mojave.

@livecode-vulcan retry 7a8fd99

livecode-vulcan commented:
😎 test success 7a8fd99f57c84120c7dc3ab8e9e48640b66813aa

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/863)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/272)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/272)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/272)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/272)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2389)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/41)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/40)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2418)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2405)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2404)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2439)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2379)
